test/model/message_test.dart

```diff
@@ -269,6 +271,80 @@ void main() {
     test('rendering-only update does not change timestamp (for old server versions)', () async {
       await checkRenderingOnly(legacy: true);
     });
+
+    group('Handle message edit state update', () {
+      final message = eg.streamMessage();
+      final otherMessage = eg.streamMessage();
+
+      Future<void> sendEvent(Message message, UpdateMessageEvent event) async {
+        await prepare();
+        await prepareMessages([message, otherMessage]);
+
+        await store.handleEvent(event);
+        checkNotifiedOnce();
+      }
+
+      test('message not moved update', () async {
+        await sendEvent(message, eg.updateMessageEditEvent(message));
+        check(store).messages[message.id].editState.equals(MessageEditState.edited);
+        check(store).messages[otherMessage.id].editState.equals(MessageEditState.none);
+      });
+
+      test('message topic moved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
+          origTopic: 'old topic',
+          newTopic:  'new topic',
+        ));
+        check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.moved)));
+      });
+
+      test('message topic resolved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
+          origTopic: 'new topic',
+          newTopic:  '✔ new topic',
+        ));
+        check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.none)));
+        await sendEvent(message, eg.updateMessageMoveEvent([message, otherMessage],
+          origTopic: '✔ new topic',
+          newTopic:  'new topic 2',
+        ));
+        check(store).messages.values.every(((message) => message.editState.equals(MessageEditState.moved)));
```

Comment:
Interesting; so it's like, 'checkmark prefix removed, and another change made', right?

Because of 'checkmark prefix removed', maybe this check makes a better pairing with the first check in the mark-_unresolved_ test?

Maybe better in any case to make a separate test case for this check. The event it's testing is for neither resolving nor unresolving a topic.